tests/support/client.rs

```diff
@@ -144,8 +175,12 @@ enum Run {
     Http2,
 }
 
-fn run(addr: SocketAddr, version: Run) -> (Sender, Running) {
-    let (tx, rx) = mpsc::unbounded::<(Request, oneshot::Sender<Result<Response, String>>)>();
+fn run(
+    addr: SocketAddr,
+    version: Run,
+    tls: Option<(Arc<ClientConfig>, DNSName)>,
```

Comment:
Nit: If we continue using a type alias for this, I'd prefer to always see `Option<TlsConfig>` rather than `Option<(...)>`
```suggestion
    tls: Option<TlsConfig>,
```